### ISSUE TITLE
@Size annotation is currently ignored on read-only collections and maps

### ISSUE THREAD

**Issue by NBau** [devopsfolks/podam#228]
If a collection is annotated with `@Size`, this should impact the number of instances generated for this collection.

**Comment by daivanov** [comment]
There is a [test](https://github.com/devopsfolks/podam/blob/develop/src/test/java/uk/co/jemos/podam/test/dto/ValidatedPojo.java#L97) [testing ](https://github.com/devopsfolks/podam/blob/develop/src/test/java/uk/co/jemos/podam/test/unit/features/validatorFramework/ValidatedPojoTest.java#L25
)`javax.validation.constraints.Size` annotation and this test is not failing.

**Comment by NBau** [comment]
Hm, that is weird. Here is the code that does not work for me (very simplified):

```
import java.util.ArrayList;
import java.util.List;

import javax.validation.constraints.Size;

import uk.co.jemos.podam.api.PodamFactory;
import uk.co.jemos.podam.api.PodamFactoryImpl;

public class Dummy {

    public static void main(String[] args){
        PodamFactory pf = new PodamFactoryImpl();
        
        Pojo p = pf.manufacturePojo(Pojo.class);
        System.out.println(p.dec.size()<=2);
    }
    
    public static class Pojo{
        
        @Size(max=2)
        List<Integer>dec;
        
        public List<Integer> getDec() {
            if (dec == null) {
                dec = new ArrayList<>();
                }
            return dec;
        }
    }
}
````
The Pojo is similar to what you get when creating Java classes from and XML Schema using JAXB, so I think this pattern should be very common. Even replacing List<Integer> with Collection<Integer> does not seem to fix it.

**Comment by daivanov** [comment]
You can try Podam's [snapshot](https://oss.sonatype.org/content/repositories/snapshots/uk/co/jemos/podam/podam/7.0.5-SNAPSHOT/podam-7.0.5-20170115.122125-1.jar).

**Comment by NBau** [comment]
Perfect. Thank you for the good support.

**Comment by daivanov** [comment]
Released in Podam 7.0.5
